### ISSUE TITLE
Warn about multiple imports of the same module

### ISSUE THREAD

**Issue by Beliavsky** [astral-sh/ruff#10638]
I request that ruff warn of multiple imports of the same module, for example

```
import numpy as np
print(np.random.normal(size=3))
import numpy as np
print(np.random.normal(size=5))
```
Thanks very much for ruff.

**Comment by MichaReiser** [comment]
Ruff's F811 already warns you about redefinition if the first definition is unused

```python
import numpy as np
import numpy as np

print(np.random.normal(size=3))

print(np.random.normal(size=5))
```

I could see this as a possible extension of [F401](https://docs.astral.sh/ruff/rules/unused-import/) that caches unused imports, assuming that re-importing can generally be considered a no-op, which I'm not sure is given. In that case it could make sense to have a separate rule (with a lower severity)

**Comment by fschulze** [comment]
Unlike with ``flake8`` the F811 rule in ``ruff`` doesn't trigger for me in a case like this:
```
import sys


if __name__ == "__main__":
    import sys
    sys.exit(0)
```

I also think it would be useful to get a new or extended rule for the original posters case.

**Comment by friday** [comment]
Thanks for pointing me here @ntBre 

I just posted in the Pylint progress issue that Ruffs implementation of F811 is marked as a substitute for pylint's `reimported` (W0404), but the Pylint implementation also detects the case already mentioned above.


Pylint's `reimported` (W0404) also detects this:

```py
import module
import module as module2
```

I'm not sure flake8 also does, but I think this is less important to cover anyway (I never had that in real code).
